perfkitbenchmarker/benchmarks/cloudsuite_web_search.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs CloudSuite Web Search benchmark.
+
+Docs:
+http://parsa.epfl.ch/cloudsuite/
+
+Runs CloudSuite Web Search to collect the statistics that show
+the operations completed per second and the minimum, maximum,
+average, 90th, and 99th response times.
+"""
+
+import posixpath
+import re
+import time
+
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+
+from perfkitbenchmarker.packages import solr521 as solr
+from perfkitbenchmarker.packages import nutch110 as nutch
+from perfkitbenchmarker.packages import faban
+
+FLAGS = flags.FLAGS
+
+BENCHMARK_INFO = {'name': 'cloudsuite_web_search',
+                  'description': 'Run CloudSuite Web Search',
+                  'scratch_disk': True,
+                  'num_machines': 3}
+
+CLOUDSUITE_WEB_SEARCH_DIR = posixpath.join(vm_util.VM_TMP_DIR,
+                                           'cloudsuite-web-search')
+
+FABAN_OUTPUT_DIR = posixpath.join(vm_util.VM_TMP_DIR, 'outputFaban')
+
+PACKAGES_URL = ('http://parsa.epfl.ch/cloudsuite/software/perfkit/web_search')
+NUTCH_SITE_URL = posixpath.join(PACKAGES_URL, 'nutch-site.xml')
+CRAWLED_URL = posixpath.join(PACKAGES_URL, 'crawl.tar.gz')
+SCHEMA_URL = posixpath.join(PACKAGES_URL, 'schema.xml')
+SEARCH_DRIVER_URL = posixpath.join(PACKAGES_URL, 'search.tar.gz')
+
+SOLR_PORT = 8983
+
+fw = None
```

Comment:
Could you avoid a global here? This won't work if the Prepare and Run/Cleanup functions are called from different runs of PerfKitBenchmarker.